mne_bids/read.py

```diff
@@ -830,15 +830,42 @@ def get_head_mri_trans(bids_path, extra_params=None, t1_bids_path=None,
     meg_bids_path.update(datatype='meg', suffix='meg')
 
     # Get the sidecar file for MRI landmarks
-    match_bids_path = meg_bids_path if t1_bids_path is None else t1_bids_path
-    t1w_path = _find_matching_sidecar(
-        match_bids_path, suffix='T1w', extension='.nii.gz')
+    t1w_bids_path = (
+        (meg_bids_path if t1_bids_path is None else t1_bids_path)
+        .copy()
+        .update(
+            datatype='anat',
+            suffix='T1w',
+            task=None
+        )
+    )
```

Comment:
this looks a bit funny, but I think I am fine with it 👍

Reply:
Yes it's ugly, I'll push a commit to improve this